share/smack/include/math/math.h

```diff
@@ -17,6 +17,27 @@
 #define fpclassify(x) (sizeof(x) == sizeof(double) ? __fpclassify(x) : __fpclassifyf(x))
 #define isfinite(x) (sizeof(x) == sizeof(double) ? __finite(x) : __finitef(x))
 
+typedef union
+{
+  float f;
+  struct
+  {
+    int sigAndExp: 31;
```

Comment:
Hmmm, you are using bit fields here. I am not sure if we support them properly in SMACK.
It would be good to first create a simple non-floating-point regression to check if bit fields actually work as expected.